elements/rh-site-status/rh-site-status.ts

```diff
@@ -0,0 +1,149 @@
+  'Partially Degraded Service': 'Partial service',
+};
+
+// map statuspage.io's statuses to icon map keys
+const statusMap: Record<string, string> = {
+  'maintenance': 'warn',
+  'critical': 'danger',
+  'major': 'warn',
+  'minor': 'warn',
+  'none': 'ok',
+};
+
+/**
+ * Website status communicates the operational status of a website or domain using a status icon and link. It is usually located in the Footer component.
+ *
+ * @summary Communicates operational status of a website or domain
+ */
+@customElement('rh-site-status')
+export class RhSiteStatus extends LitElement {
+  static readonly styles = [styles];
+
+  private static isApiStatus = (data: ApiStatus): data is ApiStatus => {
+    return (
+      typeof data === 'object' &&
+      data !== null &&
+      'status' in data &&
+      data.status !== null &&
+      'description' in data.status &&
+      'indicator' in data.status
+    );
+  };
+
+  /**
+   * Sets color context for child components, overrides parent context
+   */
+  @colorContextProvider()
+  @property({ reflect: true, attribute: 'color-palette' }) colorPalette?: ColorPalette = 'dark';
+
+  /**
+   * Sets color theme based on parent context
+   */
+  @colorContextConsumer() private on?: ColorTheme;
+
+  #logger = new Logger(this);
+
+  #text = 'Loading';
+
+  #icon: TemplateResult = html``;
+
+  #isLoading = true;
+
+  async connectedCallback() {
+    super.connectedCallback();
+    await this.#getStatus();
+  }
+
+  render() {
+    const { on = '' } = this;
+    return html`
+      <a href="https://status.redhat.com/" class="${classMap({ [on]: !!on })}">
+        ${this.#isLoading ? html`<rh-spinner size="sm"></rh-spinner>` : html`
+          <svg xmlns="http://www.w3.org/2000/svg" width="16" height="17" viewbox="0 0 16 17" fill="none">
+            ${this.#icon}
+          </svg>
+        `}
+        <span>${this.#text}</span>
+      </a>
+    `;
+  }
+
+  async #getStatus() {
+    await fetch('https://status.redhat.com/index.json', {
+      mode: 'cors',
+      cache: 'no-cache',
+      headers: {
+        Accept: 'application/json',
+      }
+    })
+      .then(response => {
+        if (response.ok) {
+          return response.json();
+        } else {
+          throw new Error(`${response.statusText}`);
+        }
+      })
+      .then((data: ApiStatus) => {
+        if (!RhSiteStatus.isApiStatus(data)) {
+          throw new Error('Invalid status data');
+        }
+        const statusText = data.status.description;
+        this.#text = textMap[statusText] || statusText;
+        this.#icon = statusIconsMap[statusMap[data.status.indicator]];
+        this.#isLoading = false;
+        this.requestUpdate();
+      })
+      .catch( error => {
+        this.#logger.warn('Error loading site status:', error);
+        this.#text = 'Error loading status';
+        this.#icon = statusIconsMap['danger'];
+        this.#isLoading = false;
+        this.requestUpdate();
+      });
+  }
```

Comment:
```suggestion
    try {
      const data: ApiStatus = await fetch('https://status.redhat.com/index.json', {
        mode: 'cors',
        cache: 'no-cache',
        headers: {
          Accept: 'application/json',
        }
      })
        .then(response => {
          if (response.ok) {
            return response.json();
          } else {
            throw new Error(`${response.statusText}`);
          }
        });
      if (!RhSiteStatus.isApiStatus(data)) {
        throw new Error('Invalid status data');
      }
      const statusText = data.status.description;
      this.#text = textMap[statusText] || statusText;
      this.#icon = statusIconsMap[statusMap[data.status.indicator]];
      this.#isLoading = false;
      this.requestUpdate();
    } catch (error) {
        this.#logger.warn('Error loading site status:', error);
        this.#text = 'Error loading status';
        this.#icon = statusIconsMap['danger'];
        this.#isLoading = false;
        this.requestUpdate();
    }
  }
```
not picking on you (i once wrote an entire blog post about why try/catch with async await is overhyped) but in this case i feel like it'll be easier to read

Reply:
we're likely to run into l10n issues with the hard coded error strings here

Reply:
> not picking on you 

All good I looked at various opinions around this.  I don't disagree with readability.  Will update.

> we're likely to run into l10n issues with the hard coded error strings her

See the TODO as well.  I pinged @kylebuch8 he doesn't know of any instance where the API right now is translated.